python/pacemaker/cts/process.py

```diff
@@ -49,3 +51,26 @@ def exit_if_proc_running(process_name):
         print("Error: %s is already running!" % process_name)
         print("Run %s only when the cluster is stopped." % sys.argv[0])
         sys.exit(ExitStatus.ERROR)
+
+
+def pipe_communicate(pipes, check_stderr=False, stdin=None):
+    """ Get text output from pipes """
+
+    if stdin is not None:
+        pipe_outputs = pipes.communicate(input=stdin.encode())
+    else:
+        pipe_outputs = pipes.communicate()
+
+    output = pipe_outputs[0].decode(sys.stdout.encoding)
+    if check_stderr:
+        output = output + pipe_outputs[1].decode(sys.stderr.encoding)
+
+    return output
+
+
+def stdout_from_command(command):
+    """ Execute command and return its standard output """
+
+    test = subprocess.Popen(shlex.split(command), stdout=subprocess.PIPE)
```

Comment:
I think we should move away from splitting command strings and pass a command array directly like the native functions, but that could be separate from this PR

Reply:
I've fixed this in stdout_from_command and its callers, and also in _start_daemons.  The other place to fix this would be where we actually add all the tests (run_cmd splits the string apart and feeds that to subprocess).  That's going to be a big, annoying, tedious patch.  So yeah that can be a followup.